src/app/courses/course-item/course-item.component.spec.ts

```diff
@@ -1,25 +1,115 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { Course } from './../../shared/courses/course.interface';
+import { DebugElement, Component } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync} from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
 
 import { CourseItemComponent } from './course-item.component';
+import { DurationPipe } from '../../duration.pipe';
+import { DatePipe } from '@angular/common';
+
+import { click } from './../../shared/helpers/spec/clickEvent';
+
+// mock the course supplied by the parent component
+const expectedCourse: Course = {
+  id: 1,
+  title: 'Course 1. This is title.',
+  dateCreation: new Date(),
+  duration: 88,
+  description: 'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.'
+};
+@Component({
+  template: `
+    <app-course-item
+      (deleteCourse)="onDelete($event)"
```

Comment:
try to follow the next common convention: `(deleteCourse)="onDeleteCourse($event)"`

Reply:
done